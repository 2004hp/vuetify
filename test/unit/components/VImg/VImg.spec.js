import { test } from '@/test'
import VImg from '@/components/VImg'

test('VImg', ({ mount }) => {
  const LOAD_FAILURE_SRC = 'LOAD_FAILURE_SRC'
  const LOAD_SUCCESS_SRC = 'LOAD_SUCCESS_SRC'

  beforeAll(() => {
    Object.defineProperty(global.Image.prototype, 'src', {
      set (src) {
        this.setAttribute('src', src)
        this._currentSrc = src
        if (src === LOAD_FAILURE_SRC) {
          setTimeout(() => this.onerror(new Error('mocked error')))
        } else {
          setTimeout(() => this.onload())
        }
      }
    })
    Object.defineProperty(global.Image.prototype, 'currentSrc', {
      get () {
        return this._currentSrc
      }
    })
  })

  it('should load', async () => {
    jest.useFakeTimers()
    const wrapper = mount(VImg, {
      propsData: { src: './image.png' }
    })

    expect(wrapper.html()).toMatchSnapshot()

    jest.runOnlyPendingTimers()
    await wrapper.vm.$nextTick()

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should emit errors', () => {
    jest.useFakeTimers()

    const wrapper = mount(VImg, {
      propsData: { src: LOAD_FAILURE_SRC }
    })

    const error = jest.fn()
    wrapper.vm.$on('error', error)

    jest.runOnlyPendingTimers()

    expect(error).toHaveBeenCalledTimes(1)
    expect(error).toHaveBeenCalledWith(LOAD_FAILURE_SRC)
    expect('Image load failed').toHaveBeenWarned()
  })

  it('should have aria attributes', () => {
    const wrapper = mount(VImg, {
      propsData: {
        src: LOAD_SUCCESS_SRC,
        alt: 'this is not a decorative image'
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })
})
