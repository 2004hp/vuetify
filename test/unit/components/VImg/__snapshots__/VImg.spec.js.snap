// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VImg should have aria attributes 1`] = `

<div role="img"
     aria-label="this is not a decorative image"
     class="v-responsive v-image"
>
  <div class="v-responsive__sizer">
  </div>
  <div class="v-image__image v-image__image--cover"
       style="background-image: url(LOAD_SUCCESS_SRC); background-position: center center;"
  >
  </div>
  <div class="v-responsive__content">
  </div>
</div>

`;

exports[`VImg should load 1`] = `

<div class="v-responsive v-image">
  <div class="v-responsive__sizer">
  </div>
  <div class="v-image__image v-image__image--cover"
       style="background-image: url(./image.png); background-position: center center;"
  >
  </div>
  <div class="v-responsive__content">
  </div>
</div>

`;

exports[`VImg should load 2`] = `

<div class="v-responsive v-image">
  <div class="v-responsive__sizer">
  </div>
  <div class="v-image__image v-image__image--cover"
       style="background-image: url(./image.png); background-position: center center;"
  >
  </div>
  <div class="v-responsive__content">
  </div>
</div>

`;
